src/workerd/api/web-socket.c++

```diff
@@ -332,7 +329,19 @@ kj::Promise<DeferredProxy<void>> WebSocket::couple(kj::Own<kj::WebSocket> other)
   auto upstream = other->pumpTo(*self);
   auto downstream = self->pumpTo(*other);
 
-  if (locality == LOCAL) {
+  auto tryGetPeer = [&]() -> kj::Maybe<WebSocket&> {
+    KJ_IF_SOME(p, peer) {
+      return p->tryGet();
+    }
+    return kj::none;
+  };
+  auto isHibernatable = [&](workerd::api::WebSocket& ws) {
+    KJ_IF_SOME(state, ws.farNative->state.tryGet<Accepted>()) {
+      return state.isHibernatable();
+    }
+    return false;
+  };
+  if (tryGetPeer() != kj::none) {
```

Comment:
Hmm, if I'm understanding correctly, prior to this commit one end of a `WebSocketPair` would be passed to `state.acceptWebSocket()`, which would set the `peer`s locality to `REMOTE`, and set its own state to `Accepted`.

Then, later on we would call `ws.couple(clientWs)`, where `ws` is the other websocket from the `WebSocketPair` (which wasn't accepted and was instead added to the `Response()`). This would mean `locality` would be `REMOTE` and we would _not_ enter this branch where we `registerPendingEvent()`.

After this change, since both ends of the `WebSocketPair()` would have a valid `peer` set (since the ctor for `WebSocketPair()` calls `setPeer()` on both) I suspect we _would_ `registerPendingEvent()` when previously we wouldn't.

I think what you really want here is to add `if (auto p = tryGetPeer(); peer != kj::none && !isHibernatable(peer)) {...}`

Reply:
Yep, you are absolutely right, and I initially did check `isHibernatable()` before calling `registerPendingEvent()`.

However, `registerPendingEvent()` has no effect for actors anyway, and I thought that this reflected the original intent of`setRemoteOnPair` better (specifically to cause deferred proxying, not necessarily to stop registering the pending event), so I decided to not put a condition in. Also, in theory I suppose we *could* implement something akin to hibernatable websockets for non-actors as well, in which case I think we *would* want to register a pending event.

Having said all that, I'm not averse to putting the `isHibernatable()` check back in if you feel strongly.

Reply:
> registerPendingEvent() has no effect for actors

Huh! I had no idea, thanks for pointing this out.